shell/platform/android/io/flutter/embedding/FlutterActivity.java

```diff
@@ -0,0 +1,236 @@
+      // Fragment so that the Intent can be delivered to any interested plugins.
+      flutterFragment.onNewIntent(intent);
+    }
+  }
+
+  @Override
+  public void onUserLeaveHint() {
+    flutterFragment.onUserLeaveHint();
+  }
+
+  /**
+   * Sets up a {@code FrameLayout} that takes up all available space in the {@code Activity}. This
+   * {@code FrameLayout} will hold a {@code FlutterFragment}, which displays a {@code FlutterView}.
+   */
+  private void createFlutterFragmentContainer() {
+    FrameLayout container = new FrameLayout(this);
+    container.setId(CONTAINER_ID);
+    container.setLayoutParams(new ViewGroup.LayoutParams(
+        ViewGroup.LayoutParams.MATCH_PARENT,
+        ViewGroup.LayoutParams.MATCH_PARENT
+    ));
+    setContentView(container);
+  }
+
+  /**
+   * If no {@code FlutterFragment} exists in this {@code FlutterActivity}, then a {@code FlutterFragment}
+   * is created and added. If a {@code FlutterFragment} does exist in this {@code FlutterActivity}, then
+   * a reference to that {@code FlutterFragment} is retained in {@code flutterFragment}.
+   */
+  private void createFlutterFragment() {
+    FragmentManager fragmentManager = getFragmentManager();
+
+    flutterFragment = (FlutterFragment) fragmentManager.findFragmentByTag(TAG_FLUTTER_FRAGMENT);
+
+    if (flutterFragment == null) {
+      // No FlutterFragment exists yet. This must be the initial Activity creation. We will create
+      // and add a new FlutterFragment to this Activity.
+      flutterFragment = FlutterFragment.newInstance(
+          isSplashScreenDesired(),
+          getInitialRoute(),
+          getAppBundlePath()
+      );
+      fragmentManager
+          .beginTransaction()
+          .add(CONTAINER_ID, flutterFragment, TAG_FLUTTER_FRAGMENT)
+          .commit();
+    }
+  }
+
+  /**
+   * Sets the color of this {@code Activity}'s status bar to a translucent black, i.e., a dark
+   * overlay.
+   *
+   * Beginning with Android Lollipop, the status bar at the top of the screen can have its color
+   * set at runtime.
+   */
+  @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+  private void configureStatusBarColor() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      Window window = getWindow();
+      window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+      window.setStatusBarColor(0x40000000);
+      window.getDecorView().setSystemUiVisibility(PlatformPlugin.DEFAULT_SYSTEM_UI);
+    }
+  }
+
+  /**
+   * Should the {@code Activity}'s {@code windowBackground} be used as a launch screen
+   * until the first frame of Flutter is rendered?
+   *
+   * This preference is controlled by setting a {@code <meta-data>} tag in the Android
+   * manifest for this {@code Activity}.
+   *
+   * TODO(mattcarroll): Should this preference be set by Intent parameter instead? How would a dev use this in production?
```

Comment:
What's the difference in production?

Reply:
I suppose it could still work this way in production, but we seem to have a very arbitrary distinction between what can be controlled via Intent, like "route", vs what can only be controlled via AndroidManifest.  Generally speaking I think everything should be controllable via Intent unless there is a good reason not to.

Reply:
Sure, triggering via Intent seems reasonable to me too.

Reply:
I expanded splash screen support to include Intent control, I also expanded initial route support to include meta data control.